snapshot_dbg_cli_tests/test_cli_services.py

```diff
@@ -0,0 +1,540 @@
+    self.user_output_class_mock = self.user_output_patcher.start()
+    self.user_output_mock = self.user_output_class_mock.return_value
+    self.addCleanup(self.user_output_patcher.stop)
+
+    self.gcloud_service_patcher = patch(
+        'snapshot_dbg_cli.cli_services.GcloudCliService', autospec=True)
+    self.gcloud_service_class_mock = self.gcloud_service_patcher.start()
+    self.gcloud_service_mock = self.gcloud_service_class_mock.return_value
+    self.addCleanup(self.gcloud_service_patcher.stop)
+
+    self.http_service_mock_patcher = patch(
+        'snapshot_dbg_cli.cli_services.HttpService', autospec=True)
+    self.http_service_class_mock = self.http_service_mock_patcher.start()
+    self.http_service_mock = self.http_service_class_mock.return_value
+    self.addCleanup(self.http_service_mock_patcher.stop)
+
+    self.permissions_service_patcher = patch(
+        'snapshot_dbg_cli.cli_services.PermissionsRestService', autospec=True)
+    self.permissions_service_class_mock = \
+        self.permissions_service_patcher.start()
+    self.permissions_service_mock = \
+        self.permissions_service_class_mock.return_value
+    self.addCleanup(self.permissions_service_patcher.stop)
+
+    self.firebase_management_service_patcher = patch(
+        'snapshot_dbg_cli.cli_services.FirebaseManagementRestService',
+        autospec=True)
+    self.firebase_management_service_class_mock = \
+        self.firebase_management_service_patcher.start()
+    self.firebase_management_service_mock = \
+        self.firebase_management_service_class_mock.return_value
+    self.addCleanup(self.firebase_management_service_patcher.stop)
+
+    self.firebase_rtdb_service_patcher = patch(
+        'snapshot_dbg_cli.cli_services.FirebaseRtdbRestService', autospec=True)
+    self.firebase_rtdb_service_class_mock = \
+        self.firebase_rtdb_service_patcher.start()
+    self.firebase_rtdb_service_mock = \
+        self.firebase_rtdb_service_class_mock.return_value
+    self.addCleanup(self.firebase_rtdb_service_patcher.stop)
+
+    self.snapshot_debugger_rtdb_service_patcher = patch(
+        'snapshot_dbg_cli.cli_services.SnapshotDebuggerRtdbService',
+        autospec=True)
+    self.snapshot_debugger_rtdb_service_class_mock = \
+        self.snapshot_debugger_rtdb_service_patcher.start()
+    self.snapshot_debugger_rtdb_service_mock = \
+        self.snapshot_debugger_rtdb_service_class_mock.return_value
+    self.addCleanup(self.snapshot_debugger_rtdb_service_patcher.stop)
+
+  def test_constructor_works_as_expected(self):
+    args = argparse.Namespace()
+
+    self.gcloud_service_mock.config_get_account = MagicMock(
+        return_value='foo@bar.com')
+    self.gcloud_service_mock.config_get_project = MagicMock(
+        return_value='cli-test-project')
+    self.gcloud_service_mock.get_access_token = MagicMock(
+        return_value='cli-test-access-token')
+
+    expected_is_debug_enabled = False
+
+    cli_services = CliServices(args)
```

Comment:
Just another request for a quick comment.  It's clear from this test that creating a CliServices object causes a bunch of stuff to happen, but something more succinct than the assertions below would be handy.  (eg. gets project and auth info from gcloud)

Reply:
Done, PTAL.